pkg/roachpb/data.go

```diff
@@ -1906,13 +1906,8 @@ func (l Lease) Equivalent(newL Lease) bool {
 	// Ignore the ReplicaDescriptor's type. This shouldn't affect lease
 	// equivalency because Raft state shouldn't be factored into the state of a
 	// Replica's lease. We don't expect a leaseholder to ever become a LEARNER
-	// replica, but that also shouldn't prevent it from extending its lease. The
-	// code also avoids a potential bug where an unset ReplicaType and a set
-	// VOTER ReplicaType are considered distinct and non-equivalent.
-	//
-	// Change this line to the following when ReplicaType becomes non-nullable:
-	//  l.Replica.Type, newL.Replica.Type = 0, 0
-	l.Replica.Type, newL.Replica.Type = nil, nil
+	// replica, but that also shouldn't prevent it from extending its lease.
+	l.Replica.Type, newL.Replica.Type = 0, 0
```

Comment:
What do zeroes mean here? Is there a named constant for what they represent?

Reply:
The zero value is `VOTER`, but I think 0 better expresses the intent: we're clearing these fields because we don't care about them for comparison purposes.